app/controllers/account.js

```diff
@@ -271,6 +275,32 @@ module.exports = function() {
                 });
             });
         },
+        remove: function(req, res) {
+            auth.authenticate(req, req.user.uid || req.user.username,
+                              data.currentPassword, function(err, user) {
+                if (err) {
+                    return res.status(400).json({
+                        status: 'error',
+                        message: 'There were problems authenticating you.',
+                        errors: err
+                    });
+                }
+                core.account.remove(req.user._id, function(err, success) {
+                    if (err) {
+                        return res.status(400).json({
+                            status: 'error',
+                            message: 'Unable to delete your profile.',
+                            errors: err
+                        });
+                    }
+
+                    // Destory session and redirect to login ('logout' action)
+                    // QUESTION: why does this redirect to the login page?
```

Comment:
I suppose because it's a page we know doesn't require you to be authenticated. (And we know you won't be after this point.)